### ISSUE TITLE
[Accessibility]: Back button after Go to Definition doesn't cause any screen reader announcements

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#35143]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33509

### Problem description

After having navigated to a page via Go to Definition, using the browser's Back functionality doesn't announce the old page/position. This results in a poor UX, as it requires a screen reader user to remember where they were at before. It's also not clear that the Back press actually worked, because there is no indication that it did.

In the context of Sourcegraph, this can be even more jarring (compared to an IDE), since one may have navigated to a different repository altogether.

Notice in the video below that the text displayed in the VoiceOver box doesn't change on going Back.

https://user-images.githubusercontent.com/93103176/167478387-71a4e3be-3749-4d2b-8778-4deba1a68dd1.mov

See also:
- https://github.com/sourcegraph/sourcegraph/issues/35094

### Expected behavior

There should be an announcement about the repository and optionally the file / line number based on the status.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by umpox** [comment]
Slightly related: https://github.com/sourcegraph/sourcegraph/issues/34724

I think at a minimum it should announce the page, I thought we would get this from the document title but unsure why that isn't being captured 🤔 

We'll look into this

**Comment by umpox** [comment]
Solved through https://github.com/sourcegraph/sourcegraph/pull/35413

Now the page will announce when navigating between files which is the primary issue here.

A sub issue might be that there is no announcement when navigating back from the panel **to the same file**. The fix here would probably be to include the current state in the URL or do a manual announcement, but that seems overkill. I think the user will still be able to understand the current page context here in that edgecase.
